### ISSUE TITLE
Keep the "Code editor" as a simple textarea

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#5698]
See also #569. The accessibility team has identified a list of 10 high priority issues for version 1 and agreed with the Gutenberg team to mark them with the "merge proposal" milestone.

The so called "text mode", i.e. a simple textarea, is the only guarantee to enable all users to publish content, regardless of the device / technology used and regardless of their level of accessibility.

The accessibility team recommendation is to keep a simple textarea available to users. If a true "Code editor" with syntax highlighting is desired, no objections to that (even if personally I'd see a dedicated block with syntax highlighting a better option). But then, there should be a third editor "mode" : a plain textarea.

**Comment by youknowriad** [comment]
That's already the case right? The current code editor is just a simple textarea. Can we close this issue?

**Comment by afercia** [comment]
If that's a confirmed decision, sure 🙂 (there were previous plans to add syntax highlighing to the code editor)

**Comment by youknowriad** [comment]
cc @jasmussen @mtias

**Comment by jasmussen** [comment]
I can only share personal opinions, I think the calls have to be made by the WordPress core leads, and secondarily the Gutenberg core leads. 

As such, take this for what it is, and not coming from a lead: but my personal opinion is that it's fine to keep the code editor completely pure. We might look at adding some plugin hooks so a plugin can add syntax highlighting, perhaps, but I'm fine with the base offering being pure. 

The past discussion, which I believe is what you're referring to Riad, surrounds the desire to add syntax highlighting to many more areas of WordPress, for example as has been added to WordPress 4.9 widgets:

![screen shot 2018-03-20 at 13 28 30](https://user-images.githubusercontent.com/1204802/37654505-9f9dbc94-2c42-11e8-8251-4dcc27897b07.png)

The closest parallel is blocks, where the Gutenberg Custom HTML block also has syntax highlighting like this. This might further extend to the "Edit as HTML" option that exists on a per block basis:

![screen shot 2018-03-20 at 13 30 35](https://user-images.githubusercontent.com/1204802/37654568-ec2e429a-2c42-11e8-82c5-f45f8ccd0443.png)

But does that extend to the full-on code editor? Like I said, I'd be fine if it didn't.

**Comment by mtias** [comment]
I'm fine with keeping the code editor as a plain text area. It can be revisited if needed as a separate project or iteration in future versions.

**Comment by afercia** [comment]
Thanks all for the clarifications. As said, no objections to add syntax highlighting to the "Code" editor (or to other places) but then at that point there should be a "pure" fallback, i.e. 3 modes:
1. Visual: visual editor
2. Code: syntax highlighting
3. "Plain": simple textarea
